engine/common/rpc/convert/executor_metadata.go

```diff
@@ -0,0 +1,29 @@
+package convert
+
+import (
+	"github.com/onflow/flow/protobuf/go/flow/entities"
+
+	"github.com/onflow/flow-go/model/access"
+)
+
+func ExecutorMetadataToMessage(metadata *access.ExecutorMetadata) (*entities.ExecutorMetadata, error) {
+	if metadata == nil {
+		return nil, ErrEmptyMessage
+	}
+
+	return &entities.ExecutorMetadata{
+		ExecutionResultId: IdentifierToMessage(metadata.ExecutionResultID),
+		ExecutorId:        IdentifiersToMessages(metadata.ExecutorIDs),
+	}, nil
+}
+
+func MessageToExecutorMetadata(metadata *entities.ExecutorMetadata) (*access.ExecutorMetadata, error) {
+	if metadata == nil {
+		return nil, ErrEmptyMessage
```

Comment:
similarly, since it is valid for metadata to be nil in the response, this should allow nil and just return an empty `access.ExecutorMetadata`